src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -10,25 +14,78 @@
     private static final Logger log = LoggerFactory.getLogger(ExecutionStubImpl.class);
 
     /**
+     * Sets up an executor with some sane defaults that both executeCommand
+     * implementations depend on.
+     *
+     * @param errStream stream to write any errors during command execution
+     * @param outStream stream to write any output during command execution
+     * @return Executor with exit-code checking disabled and output and error streams setup
+     */
+    private static Executor setupExecutor(ByteArrayOutputStream errStream, ByteArrayOutputStream outStream) {
+        DefaultExecutor executor = new DefaultExecutor();
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outStream, errStream);
+
+        // Don't throw exception on non-zero exit code
+        executor.setExitValues(null);
+
+        // Set up the handlers
+        executor.setStreamHandler(streamHandler);
+
+        return executor;
+    }
+
+    /**
      * Executes the given command in a separate process.
      *
      * @param command the command to execute
-     * @return A 2-element array; the first element is the output of the process.
-     *      The second element is the exit code of the process.
+     * @return An ExecutionStubResult with output, error, and the exit code of the process
      *
      * @throws InterruptedException
      * @throws IOException
      */
     public static ExecutionStubResult executeCommand(String command)
             throws InterruptedException, IOException {
+        org.apache.commons.exec.CommandLine commandLine = org.apache.commons.exec.CommandLine.parse(command);
+        ByteArrayOutputStream errStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+
+        Executor executor = setupExecutor(errStream, outStream);
```

Comment:
I think you can push the rest of the code in this method below this line into `setupExecutor`, since it's the same in both versions of `executeCommand`.  In which case, `setupExecutor` probably warrants a rename, but it'd be a little DRY-er that way.